yt/visualization/volume_rendering/tests/test_save_render.py

```diff
@@ -0,0 +1,39 @@
+import os
+import tempfile
+import shutil
+import yt
+from yt.testing import fake_random_ds
+from unittest import TestCase
+
+def setup():
+    """Test specific setup."""
+    from yt.config import ytcfg
+    ytcfg["yt", "__withintesting"] = "True"
+
+
+class SaveRenderTest(TestCase):
+    # This toggles using a temporary directory. Turn off to examine images.
+    use_tmpdir = True
+
+    def setUp(self):
+        if self.use_tmpdir:
+            self.curdir = os.getcwd()
+            # Perform I/O in safe place instead of yt main dir
+            self.tmpdir = tempfile.mkdtemp()
+            os.chdir(self.tmpdir)
+        else:
+            self.curdir, self.tmpdir = None, None
+
+    def tearDown(self):
+        if self.use_tmpdir:
+            os.chdir(self.curdir)
+            shutil.rmtree(self.tmpdir)
+
+    def test_save_render(self):
+        ds = fake_random_ds(32)
+        sc = yt.create_scene(ds)
+        sc.save('raw.png') # will use render = True by default
+        sc.save('clip_2.png', sigma_clip=2, render=False) # will pull render
+        sc.save('clip_4.png', sigma_clip=4.0, render=False)
+
```

Comment:
Shouldn't we compare the images to check they are indeed different ?

Reply:
Sure! That would make sense -- then it becomes an actual test of functionality, not just a test of keyword arguments... 

is there a way you'd recommend to do this? Load the images back in and use `yt.testing.assert_equal` or `assert_almost_equal`? 

In terms of loading the image back in -- is there a "yt" way to do it? I could use `matplotlib.pyplot.imread` but maybe there's a more appropriate method?

Reply:
Yes there's a "yt way" to run image comparison tests.
Unfortunately, I can't remember off the top of my head... but I know @munkm does !
In the mean time, you may want to read this https://yt-project.org/docs/dev/developing/testing.html#answer-testing

Reply:
Ah, yes, I need to re-read that :) `How to Write Image Comparison Tests` has a description of using `GenericImageTest`

Reply:
Though in this case I need to check that the 3 saved rendered images are NOT equal. But I'll read through that and figure it out.

Reply:
ah yes, that's the one I meant to link, sorry !
I remember I was quite confused last time I had to do something with it, but @munkm's help was extremely useful

Reply:
> Though in this case I need to check that the 3 saved rendered images are NOT equal. But I'll read through that and figure it out.

Well the easy way out is to check they are different when you first run it, store the 3 outputs and then compare against those in the future.